lib/lib/network/request/request_handler.dart

```diff
@@ -0,0 +1,19 @@
+import 'dart:async';
+
+import 'package:f1_application/lib/datamanagement/error_handler.dart';
+import 'package:f1_application/lib/model/driver.dart';
+import 'package:f1_application/lib/model/season.dart';
+
+class RequestHandler{
+  Future<dynamic> _handle(request) async {
+    try {
+      return await request();
+    } catch (e) {
+      throw ErrorHandler().handle(e);
+    }
+  }
```

Comment:
Ez így teljesen jó, az egyetlen amit változtatnék rajta hogy a `dynamic` az egy túl általános return típus és generikus függvénnyel jobb volna ezt megcsinálni, hogy kisebb eséllyel lehessen típusproblémákba belefutni. Ahogy nézem már most is lekezeli a fordító a lenti 3 fügvénnyel, de azok nem kellenek ha így definiálod a `handle` függvényt

```
class RequestHandler{
  Future<T> handle<T>(Function request) async {
    try {
      return await request();
    } catch (e) {
      throw ErrorHandler().handle(e);
    }
  }
}
```